tests/cli/test_list.py

```diff
@@ -1,96 +1,147 @@
 import json
+import pathlib
+from unittest.mock import patch
 
 import pytest
 
 from pdm.cli import actions
-from pdm.exceptions import PdmException
+from pdm.cli.commands.list import Command
+from pdm.models.specifiers import PySpecSet
+from pdm.utils import path_to_url
+from tests import FIXTURES
+from tests.conftest import Distribution
 
 
 def test_list_command(project, invoke, mocker):
-    do_list = mocker.patch.object(actions, "do_list")
-    invoke(["list"], obj=project)
-    do_list.assert_called_once()
+    # Calls the correct handler within the Command
+    with patch.object(Command, "handle_list") as m:
```

Comment:
```suggestion
    with mocker.patch.object(Command, "handle_list") as m:
```

Reply:
I can't get the mocker fixture to work at all, not sure why yet!  Will take another look tomorrow.

Reply:
Sorted 👍